xbmc/games/ports/PortMapper.cpp

```diff
@@ -23,13 +23,18 @@
 #include "peripherals/devices/Peripheral.h"
 #include "peripherals/Peripherals.h"
 
+#include <assert.h>
+
 using namespace KODI;
 using namespace GAME;
 using namespace JOYSTICK;
 using namespace PERIPHERALS;
 
-CPortMapper::CPortMapper()
+CPortMapper::CPortMapper(PERIPHERALS::CPeripherals* peripheralManager) :
+  m_peripheralManager(peripheralManager)
 {
+  assert(m_peripheralManager != nullptr);
```

Comment:
If null doesn't make sense here, should we pass by reference rather than pointer?